instrumentation/rack/lib/opentelemetry/instrumentation/rack/middlewares/event_handler.rb

```diff
@@ -52,7 +53,11 @@ class EventHandler
           # @param [Rack::Response] This is nil in practice
           # @return [void]
           def on_start(request, _)
-            return if untraced_request?(request.env)
+            if untraced_request?(request.env)
+              ctx = OpenTelemetry::Common::Utilities.untraced
+              request.env[UNTRACED_CTX_TOKEN] = OpenTelemetry::Context.attach(ctx)
```

Comment:
Why do we need another context (attached) and another token for this? We should add a `context = OpenTelemetry::Context.current` parameter to `extract_remote_context` and pass in the untraced context if necessary. I.e.:
```ruby
parent_context = if untraced_request?(request.env)
    extract_remote_context(request, OpenTelemetry::Common::Utilities.untraced)
  else
    extract_remote_context(request)
  end
```